launcher/src/main/java/org/apache/spark/launcher/CommandBuilderUtils.java

```diff
@@ -334,6 +334,18 @@ static void addPermGenSizeOpt(List<String> cmd) {
   }
 
   /**
+   * Gets the OutOfMemoryError option for Spark if the user hasn't set it.
+   */
+  public static String getOutOfMemoryErrorArgument(List<String> cmd) {
```

Comment:
I don't see that? the only similar method I see manages `-XX:MaxPermGen` and it does not work this way.